src/main/java/io/github/amerebagatelle/fabricskyboxes/util/object/AnimationTextures.java

```diff
@@ -0,0 +1,65 @@
+package io.github.amerebagatelle.fabricskyboxes.util.object;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.util.Identifier;
+
+import java.util.List;
+
+public class AnimationTextures {
+    public static final Codec<AnimationTextures> CODEC = RecordCodecBuilder.create(instance -> instance.group(
+            Identifier.CODEC.listOf().fieldOf("north").forGetter(AnimationTextures::getNorth),
```

Comment:
Can these be a list of `Textures` instead?

Reply:
Using a list of `Textures` would result in a larger JSON format, I personally think it's much easier for the resource pack maker to use `AnimationTextures`.

Reply:
You could make a serialization of the AnimationTextures to a list of Textures?

Reply:
it’s safer because you don’t need to make sure the size of each list is the same